test_rclcpp/test/test_timer.cpp

```diff
@@ -108,16 +109,15 @@ TEST(CLASSNAME(test_time, RMW_IMPLEMENTATION), timer_during_wait) {
   ASSERT_EQ(0, counter);
 
   auto spinner =
-    [&executor, &node]() -> void
+    [&executor]() -> void
     {
       printf("spin() until shutdown\n");
-      executor.add_node(node, false);
       executor.spin();
     };
 
   auto thread = std::thread(spinner);
-  printf("sleeping for 4 periods\n");
-  std::this_thread::sleep_for(4 * period);
+  printf("sleeping for 3.5 more periods\n");
+  std::this_thread::sleep_for(std::chrono::duration_cast<std::chrono::nanoseconds>(3.5 * period));
```

Comment:
Why 3.5? In combination with the 0.5 sleep earlier that make it align exactly with a timer event and therefore very likely undeterministic.